### DISCUSSION TITLE
status code 404 when creating a registration token for a repository

### DISCUSSION THREAD

**Discussion by shawnzhu** [community/community#43524] [Enterprise] [answered]
### Select Topic Area

Question

### Body

I used my personal access token to register self-hosted runners:

```
curl \
  -X POST \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer ${ACCESS_TOKEN}"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/${ORG_NAME}/${REPO_NAME}/actions/runners/registration-token
```

I got:

```
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/actions#create-a-registration-token-for-a-repository"
}
```

The problem is it works (status code `201` with working token) for repos I am admin. does it mean only admin role could register self-hosted runners? At org level, what role is required to be able to register self-hosted runners?

**Comment by MosJef** [comment, marked as answer]
In the [About self-hosted runners in GitHub Actions](https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28) situated at the very top of the page it is stated:

_Authenticated users must have admin access to repositories or organizations, or the `manage_runners:enterprise` scope for enterprises to use these endpoints._

**Reply by shawnzhu** [reply]
Thank you for the key information. it is the status code `404` is not helpful. but I could understand why it is not a `403`
